pkg/runtime/openshift.go

```diff
@@ -1,5 +1,134 @@
 package runtime
 
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift/ansible-service-broker/pkg/clients"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
 func (o openshift) getRuntime() string {
 	return "openshift"
 }
+
+func (o openshift) shouldJoinNetworks() (bool, PostSandboxCreate, PostSandboxDestroy) {
+	ocli, err := clients.Openshift()
+	if err != nil {
+		log.Errorf("unable to get openshift client - %v", err)
+		// Defaulting if anything goes wrong to not join the networks.
+		return false, nil, nil
+	}
+	pluginName, err := ocli.GetClusterNetworkPlugin()
+	log.Debugf("plugin for the network - %v", pluginName)
+	if err != nil {
+		// The plugins could not be defined. ex: when using oc cluster up
+		// or a pure k8s cluster. Therefore making this a notice.
+		log.Noticef("unable to retrieve the network plugin, defaulting to not joining networks - %v", err)
+		// Defaulting to not join the networks.
+		return false, nil, nil
+	}
+
+	// Case insensitive check here because want to prepare if things change.
+	if strings.ToLower(pluginName) == "redhat/openshift-ovs-multitenant" {
+		log.Debugf("stating that the pluginname is multitenant - %v", pluginName)
+		return true, addPodNetworks, isolatePodNetworks
+	}
+	return false, nil, nil
+}
+
+func addPodNetworks(pod, ns string, targetNS []string, apbRole string) error {
+	log.Debugf("adding pod networks together namespace: %v, target namespaces: %v", ns, targetNS)
+	// Check to make sure that we have a target namespace.
+	if len(targetNS) < 1 {
+		return fmt.Errorf("Can not find target namespace to add to its networ")
+	}
+	o, err := clients.Openshift()
+	if err != nil {
+		return err
+	}
+	// Get corresponding NetNamespace for given namespace
+	netns, err := o.GetNetNamespace(ns)
+	if err != nil {
+		return err
+	}
+	_, err = o.JoinNamespacesNetworks(netns, targetNS[0])
+	if err != nil {
+		log.Errorf("Unable to join netns: %v to targetNS: %v", netns.Name, targetNS[0])
+		return err
+	}
+
+	//  wait for some time, to determine if the change was applied correctly.
+	backoff := wait.Backoff{
+		Steps:    15,
+		Duration: 500 * time.Millisecond,
+		Factor:   1.1,
+	}
+	return wait.ExponentialBackoff(backoff, func() (bool, error) {
+		return didAnnotationUpdate("join", netns.NetName)
+	})
```

Comment:
as mentioned above, this could fail if the ClusterNetwork object is stale and mistakenly claims that they're using multitenant when really they're using a third-party plugin now